### DISCUSSION TITLE
Inconsistent behaviour after clearing form fields

### DISCUSSION THREAD

**Discussion by jira-zz** [react-hook-form/react-hook-form#11357] [Q&A] [answered]
I have a form with password and confirm password fields, The passwords are optional.
I have observed the following behavior:
- fill the password fields with different values
- it displays error message as expected
- delete first the confirm password field and then the password field
- both fields are now empty, but the error message stays

If, however, you first delete the password and then the confirm password then the error message correctly disappears.

[Sandbox](https://codesandbox.io/p/devbox/rhf-confirm-password-j95pdr)

**Comment by Moshyfawn** [comment, marked as answer]
Validation is localized to the edited field. Use [the `deps` prop](https://arc.net/l/quote/xkxildjz) to run checks for all dependent fields when one is updated.

Here's [a working codesandbox](https://codesandbox.io/p/devbox/rhf-confirm-password-j95pdr?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clqxxl5sp00073b6fvn2vjjbk%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clqxxl5so00023b6f0wadf08y%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clqxxl5so00043b6fgnlhpt34%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clqxxl5sp00063b6fwxwbyqjc%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clqxxl5so00023b6f0wadf08y%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqxxl5so00013b6f01kwadzi%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Ftsconfig.json%2522%257D%255D%252C%2522id%2522%253A%2522clqxxl5so00023b6f0wadf08y%2522%252C%2522activeTabId%2522%253A%2522clqxxl5so00013b6f01kwadzi%2522%257D%252C%2522clqxxl5sp00063b6fwxwbyqjc%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqxxl5sp00053b6fkmmikpn2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clqxxl5sp00063b6fwxwbyqjc%2522%252C%2522activeTabId%2522%253A%2522clqxxl5sp00053b6fkmmikpn2%2522%257D%252C%2522clqxxl5so00043b6fgnlhpt34%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqxxl5so00033b6f0m69vndb%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%255D%252C%2522id%2522%253A%2522clqxxl5so00043b6fgnlhpt34%2522%252C%2522activeTabId%2522%253A%2522clqxxl5so00033b6f0m69vndb%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D).
